src/hooks/useDeepCompareEffect.ts

```diff
@@ -0,0 +1,14 @@
+import React, { useEffect, useRef } from 'react';
+import isEqual from 'lodash/isEqual';
+import cloneDeep from 'lodash/cloneDeep';
+
+export const useDeepCompareEffect = (effect: React.EffectCallback, deps: any) => {
+  const ref = useRef<any>();
+  useEffect(() => {
+    if (isEqual(ref.current, deps)) {
+      return;
+    }
+    ref.current = cloneDeep(deps);
```

Comment:
之前遇到过一个 `cloneDeep` 引起的性能问题，在 全局配置的 hook 里面写了 `cloneDeep`，导致 IE 非常卡顿，当时一堆人查了好几天。

此处使用，需慎重。每一次的变化都是一次 `cloneDeep`，使用本身就有性能问题的函数，来实现 `use deep compare to avoid some useless change`，是否真的有效益。有使用前后的收益对比吗？